src/player.cpp

```diff
@@ -10884,42 +10884,45 @@ bool player::has_enough_charges( const item &it, bool show_msg ) const
     return true;
 }
 
-bool player::consume_charges(item *used, long charges_used)
+bool player::consume_charges( item& used, long qty )
 {
-    // Non-tools can use charges too - when they're comestibles
-    const auto tool = dynamic_cast<const it_tool*>(used->type);
-    const auto comest = dynamic_cast<const it_comest*>(used->type);
-    if( charges_used <= 0 || (tool == nullptr && comest == nullptr) ) {
+    if( !has_item( &used ) ) {
+        debugmsg( "Tried to consume charges for an item not in players possession" );
```

Comment:
`consume_charges` is called from `player::invoke_item`, which in turn is called from several places (bionics, pickup.cpp) with a temporary item as argument. The item (being temporary) is not in the players possession and triggers this debug message. Try to use the hotplate of a kitchen unit for example.